keyconcepts.md

```diff
@@ -1,5 +1,18 @@
 Just as an initial draft for the key concepts:
 
+We have to prioritize those two when we code the problem-solving mindset:
+1- Systems thinking (System theory in general)
+2- Minimalism.
+
+Because all of the others stem from those two. Even a problem or a solution can be defined as a "system" that consists of interrelated subsystems. 
+
+For instance, a problem is a system that becomes an obstacle against the existence or stability of another system.
+
+Minimalism is the design strategy of problem-solving as minimalism makes it easier to detect and fix errors; enhance or refactor subsystems.
+
+(Related to minimalism: Law of Demeter (the principle of least knowledge), the principle of least privilege, the principle of least effort, the principle of least action.)
```

Comment:
Law of Demeter also reminded me of the usage of private variables in object oriented programming. While working with some object of some class, dealing with private variables, whose only purpose is to do something inside the class, could have led to unwanted reference issues and chains and would have been overall tedious, if they weren't declared them as private.

Reply:
You are absolutely right.
Nevertheless, what you mention is also one of the proofs that our minds work alike because I also thought the same when I first read about the Law of Demeter.
 
It is also related to our idea of "Spiral" and the question: "Which people should we prioritize to share our love and information."